### PR TITLE
[SofaCUDA] Rework and move benchmark scenes for HexahedronFEMForceField and TetrahedronFEMForceField

### PR DESCRIPTION
**Pull request by epernod**
- Add performances benchmarks for gpu vs cpu in SofaCUDA/scenes/benchmarks/
  - TetrahedronFEMForceField_beam10x10x40_cpu.scn  (~23.5 fps)
  - TetrahedronFEMForceField_beam10x10x40_gpu.scn  (~158 fps)
  - TetrahedronFEMForceField_beam16x16x76_cpu.scn  (~8 fps)
  - TetrahedronFEMForceField_beam16x16x76_gpu.scn  (~148 fps)

  - HexahedronFEMForceField_beam10x10x40_cpu.scn  (~30 fps)
  - HexahedronFEMForceField_beam10x10x40_gpu.scn  (~102 fps)
  - HexahedronFEMForceField_beam16x16x76_cpu.scn  (~9.5 fps)
  - HexahedronFEMForceField_beam16x16x76_gpu.scn  (~42 fps)

- Add cpu-gpu validations scenes in SofaCUDA/scenes/cpu-gpu_validation/  with green beam for gpu and red for cpu
  - CudaTetrahedronFEMForceField_beam10x10x40_implicit.scn
  - CudaTetrahedronFEMForceField_beam10x10x40_rk4.scn
  
  - CudaHexahedronFEMForceField_beam10x10x40_implicit.scn
  - CudaHexahedronFEMForceField_beam10x10x40_rk4.scn

- Remove all old scenes for hexa and tetra fem in SofaCUDA/examples/
______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: applications/plugins/SofaCUDA/scenes/benchmarks/HexahedronFEMForceField_beam16x16x76_gpu.scn [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        <HexahedronFEMForceField name="FEM" template="CudaVec3f" youngModulus="1000" poissonRatio="0.3" method="large" />
+        <PlaneForceField normal="0 1 0" d="2" stiffness="10000"  showPlane="1" />
+        
+        <Node name="surface">
+            <QuadSetTopologyContainer name="Container" />
+            <QuadSetTopologyModifier name="Modifier" />
+            
+            <Hexa2QuadTopologicalMapping input="@../Container" output="@Container" />
+            <Node name="Visu">
+                <OglModel name="Visual" color="green" />
```
Comment (alxbilger) [inline applications/plugins/SofaCUDA/scenes/benchmarks/HexahedronFEMForceField_beam16x16x76_gpu.scn:41]: I don't know the differences, but there is also `CudaVisualModel` which seems specialized to render Cuda dofs. IMO it avoids host/device copies.

File: applications/plugins/SofaCUDA/scenes/benchmarks/TetrahedronFEMForceField_beam10x10x40_gpu.scn [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        <TetrahedronFEMForceField template="CudaVec3f" name="FEM" computeGlobalMatrix="false" method="large" poissonRatio="0.3" youngModulus="2000" />
+        
+        <Node name="surface">
+            <TriangleSetTopologyContainer name="Container" />
+            <TriangleSetTopologyModifier name="Modifier" />
+            <TriangleSetGeometryAlgorithms template="CudaVec3f" name="GeomAlgo" />
+            
+            <Tetra2TriangleTopologicalMapping input="@../Container" output="@Container" />
+            <Node name="Visu">
+                <OglModel name="Visual" color="green" />
```
Comment (alxbilger) [inline applications/plugins/SofaCUDA/scenes/benchmarks/TetrahedronFEMForceField_beam10x10x40_gpu.scn:50]: Probably `CudaVisualModel` again

File: applications/plugins/SofaCUDA/scenes/benchmarks/TetrahedronFEMForceField_beam16x16x76_gpu.scn [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+		<PlaneForceField normal="0 1 0" d="2" stiffness="10000"  showPlane="1" />
+        
+        <Node name="surface">
+            <TriangleSetTopologyContainer name="Container" />
+            <TriangleSetTopologyModifier name="Modifier" />
+            <TriangleSetGeometryAlgorithms template="CudaVec3f" name="GeomAlgo" />
+            
+            <Tetra2TriangleTopologicalMapping input="@../Container" output="@Container" />
+            <Node name="Visu">
+                <OglModel name="Visual" color="green" />
```
Comment (alxbilger) [inline applications/plugins/SofaCUDA/scenes/benchmarks/TetrahedronFEMForceField_beam16x16x76_gpu.scn:53]: Probably `CudaVisualModel` again

File: applications/plugins/SofaCUDA/scenes/cpu-gpu_validation/CudaHexahedronFEMForceField_beam10x10x40_implicit.scn [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        <FixedConstraint indices="@ROI1.indices" />
+        <HexahedronFEMForceField name="FEM" template="CudaVec3f" youngModulus="2000" poissonRatio="0.3" method="large" />
+
+        <Node name="surface">
+            <QuadSetTopologyContainer name="Container" />
+            <QuadSetTopologyModifier name="Modifier" />
+            
+            <Hexa2QuadTopologicalMapping input="@../Container" output="@Container" />
+            <Node name="Visu">
+                <OglModel name="Visual" color="green" />
```
Comment (alxbilger) [inline applications/plugins/SofaCUDA/scenes/cpu-gpu_validation/CudaHexahedronFEMForceField_beam10x10x40_implicit.scn:42]: Probably `CudaVisualModel` again

File: applications/plugins/SofaCUDA/scenes/cpu-gpu_validation/CudaHexahedronFEMForceField_beam10x10x40_rk4.scn [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        <FixedConstraint indices="@ROI1.indices" />
+        <HexahedronFEMForceField name="FEM" template="CudaVec3f" youngModulus="2000" poissonRatio="0.3" method="large" />
+
+        <Node name="surface">
+            <QuadSetTopologyContainer name="Container" />
+            <QuadSetTopologyModifier name="Modifier" />
+            
+            <Hexa2QuadTopologicalMapping input="@../Container" output="@Container" />
+            <Node name="Visu">
+                <OglModel name="Visual" color="green" />
```
Comment (alxbilger) [inline applications/plugins/SofaCUDA/scenes/cpu-gpu_validation/CudaHexahedronFEMForceField_beam10x10x40_rk4.scn:41]: Probably `CudaVisualModel` again

File: applications/plugins/SofaCUDA/scenes/cpu-gpu_validation/CudaTetrahedronFEMForceField_beam10x10x40_implicit.scn [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        <TetrahedronFEMForceField template="CudaVec3f" name="FEM" computeGlobalMatrix="false" method="large" poissonRatio="0.3" youngModulus="2000" />
+        
+        <Node name="surface">
+            <TriangleSetTopologyContainer name="Container" />
+            <TriangleSetTopologyModifier name="Modifier" />
+            <TriangleSetGeometryAlgorithms template="CudaVec3f" name="GeomAlgo" />
+            
+            <Tetra2TriangleTopologicalMapping input="@../Container" output="@Container" />
+            <Node name="Visu">
+                <OglModel name="Visual" color="green" />
```
Comment (alxbilger) [inline applications/plugins/SofaCUDA/scenes/cpu-gpu_validation/CudaTetrahedronFEMForceField_beam10x10x40_implicit.scn:52]: Probably `CudaVisualModel` again

File: applications/plugins/SofaCUDA/scenes/cpu-gpu_validation/CudaTetrahedronFEMForceField_beam10x10x40_rk4.scn [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+        <TetrahedronFEMForceField template="CudaVec3f" name="FEM" computeGlobalMatrix="false" method="large" poissonRatio="0.3" youngModulus="2000" />
+        
+        <Node name="surface">
+            <TriangleSetTopologyContainer name="Container" />
+            <TriangleSetTopologyModifier name="Modifier" />
+            <TriangleSetGeometryAlgorithms template="CudaVec3f" name="GeomAlgo" />
+            
+            <Tetra2TriangleTopologicalMapping input="@../Container" output="@Container" />
+            <Node name="Visu">
+                <OglModel name="Visual" color="green" />
```
Comment (alxbilger) [inline applications/plugins/SofaCUDA/scenes/cpu-gpu_validation/CudaTetrahedronFEMForceField_beam10x10x40_rk4.scn:51]: Probably `CudaVisualModel` again

**Review by alxbilger — COMMENTED** [review]
I like the idea of this PR. Good job!

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

**Pull request merged**